src/nodes/utils/SetNode.js

```diff
@@ -30,7 +71,9 @@ class SetNode extends TempNode {
 		const { sourceNode, components, targetNode } = this;
 
 		const sourceType = this.getNodeType( builder );
-		const targetType = builder.getTypeFromLength( components.length, targetNode.getNodeType( builder ) );
+
+		const componentType = builder.getComponentType( targetNode.getNodeType( builder ) );
+		const targetType = builder.getTypeFromLength( components.length, componentType );
```

Comment:
@sunag FYI: There is a small fix to ensure the `targetType` is computed. Previously, something like `color.setRG( vec2( 1, 1 ) )` produced a runtime error.